### PR TITLE
task-driver: integration: update-wallet: Add additional tests with ERC20 deposit + withdrawal

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR adds additional `update-wallet` tests including:
- Adding and removing fees
- Depositing and withdrawing balances from the pool. This makes use of the pre-deployed `ETH` ERC-20 on the katana devnet node

### Testing
- Unit and integration tests pass

### CODE REVIEW

File: task-driver/integration/tests/update_wallet.rs [hunk 1]
```diff
@@ -47,57 +71,233 @@ async fn test_update_wallet_then_recover(test_args: IntegrationTestArgs) -> Resu
...
     let success = handle.await?;
-    assert_true_result!(success)?;
+    assert_true_result!(success)
+}
 
-    // Now attempt to lookup the wallet and verify its construction
-    lookup_wallet_and_check_result(&wallet, blinder_seed, share_seed, test_args).await
+/// Execute a wallet update, then lookup the new wallet from on-chain state and verify it has
+/// been correctly constructed
+async fn execute_wallet_update_and_verify(
```
Comment (akirillo) [inline task-driver/integration/tests/update_wallet.rs:79]: tiny nit: this name is a bit vague, could we do `execute_wallet_update_and_verify_shares`?

File: task-driver/integration/tests/update_wallet.rs [hunk 2]
```diff
@@ -47,57 +71,233 @@ async fn test_update_wallet_then_recover(test_args: IntegrationTestArgs) -> Resu
...
-        client.clone(),
-        test_args.network_sender.clone(),
-        test_args.global_state.clone(),
-        test_args.proof_job_queue.clone(),
-    )?;
+        Some(ExternalTransfer {
+            mint: mint.clone(),
+            amount: amount.into(),
+            account_addr: account_addr.clone(),
+            direction: ExternalTransferDirection::Deposit,
```
Comment (akirillo) [inline task-driver/integration/tests/update_wallet.rs:275]: thoughts on making this a `lazy_static` item as well? i.e. a `DUMMY_DEPOSIT` and a `DUMMY_WITHDRAWAL`?

**Review by akirillo — APPROVED** [review]
just smol nits, nothing blocking - lgtm!

File: task-driver/integration/tests/update_wallet.rs (hunk 2 above)
Comment (joeykraut) [reply task-driver/integration/tests/update_wallet.rs:275]: It's parameterized by a CLI arg, so it isn't static

**Pull request merged**